### PR TITLE
Refactore of the build system

### PR DESCRIPTION
**Pull request by tdrozdovsky**
Signed-off-by: Taras Drozdovskyi <t.drozdovsky@samsung.com>

# Description

After request from external developer on supporting the `Alpine` container instead of `Ubuntu` [Vitalii55' branch](https://github.com/Vitalii55/edge-home-orchestration-go/tree/add-alpine-linux-support), it became obvious to me that we need to configure a more flexible system for configuring the build's parameters. I suggest abandoning the management of the build using the `build.sh` script, and fully use `Makefile`. For convenience, the configuration is used standard tool `Kconfig` and its configuration files. This is the first step to consider my proposal by the community. I am very interested in your opinion on this issue. 

## Type of change

- [x] Code cleanup/refactoring
- [x] Documentation update

# How Has This Been Tested?

For build:
```
$ make distclean
$ make create_context CONFIGURATION_FILE_NAME=x86_64c
$ make
```
for run:
```
$ make run
```

**Test Configuration**:
* Firmware version: Ubuntu 18.04, etc.)
* Hardware: x86-64, ARM
* Toolchain: Docker and Go recommended versions
* Edge Orchestration Release: Coconut

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	---help---
+		"MNEDC Client"
+
+endchoice
+
+config SECURE_MODE
+	bool "Secure Mode is enable"
+    default y
+	---help---
+		"Secure mode is enable"
```
Comment (t25kim) [inline Kconfig:84]: Spaces and tabs are mixed. Can you unify the indentation method?

File: Makefile [hunk 2]
```diff
@@ -107,14 +191,89 @@ clean:
...
+	$(call build-object-c)
+	$(call build-result)
+else ifeq ($(CONFIG_ANDROID),y)
+	$(call print_header, "Target Binary is for Android")
+	$(call print_header, "Create Android archive from Java interface")
+	make build-object-java
+	$(call build-result)
+endif
+
+.config:
```
Comment (t25kim) [inline Makefile:244]: Is it possible to copy the config file automatically based on the kernel information when I run "make create_context" only without `CONFIGURATION_FILE_NAME`?

**Review by t25kim — COMMENTED** [review]
This is great!
Please take a look my comments.

File: docs/platforms/x86_64_linux/x86_64_native.md [hunk 3]
```diff
@@ -44,6 +49,9 @@ test
     ├── copy_srvs.sh
     └── ls_srv
         └── ls_srv.conf
+tools
+└── create_fs.sh 
+
```
[the remark this answers is not in the document]
Comment (tdrozdovsky) [reply docs/platforms/x86_64_linux/x86_64_native.md]: Done! Thank you!

File: Kconfig (hunk 1 above)
Comment (tdrozdovsky) [reply Kconfig]: Done! Thank you!

**Review by t25kim — APPROVED** [review]
LGTM. Thanks!

**Review by MoonkiHong — APPROVED** [review]
LGTM. Thank you! 😄

**Comment by MoonkiHong** [comment]
@tdrozdovsky LGTM. Believe that no more task is left from your side, right? Let us know.

**Comment by tdrozdovsky** [comment]
> 
> 
> @tdrozdovsky LGTM. Believe that no more task is left from your side, right? Let us know.

@MoonkiHong It's all! ))).

**Pull request merged**